script/security-check.js

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable no-console */
+/*
+ * This script parses the yarn audit JSON results to find security advisories
+ * that affect modules in our dependencies list that are moderate or higher
+ * and aren't in our exceptions list
+ */
+const fs = require('fs');
+const path = require('path');
+const { spawn } = require('child_process');
+
+const packageJSON = require('../package.json');
+
+const exceptions = JSON.parse(
+  fs.readFileSync(path.join(__dirname, '../.nsprc'), 'utf8'),
+).exceptions;
+
+const dependencySet = new Set(Object.keys(packageJSON.dependencies));
+const exceptionSet = new Set(exceptions);
+const severitySet = new Set(['high', 'critical', 'moderate']);
+
+function getAffectedModule(data) {
+  return data.resolution.path.split('>')[0];
+}
+
+function processAuditResults(audit) {
+  // The output isn't valid JSON, it's a list of JSON
+  // objects separated by newlines
+  const advisories = audit
+    .trim()
+    .split('\n')
+    .map(a => JSON.parse(a));
+
+  const validAdvisories = advisories.filter(adv => {
+    if (adv.type === 'auditAdvisory') {
+      const advisoryData = adv.data.advisory;
+      const affectedModule = getAffectedModule(adv.data);
+      return (
+        !exceptionSet.has(advisoryData.url) &&
+        severitySet.has(advisoryData.severity) &&
+        dependencySet.has(affectedModule)
```

Comment:
Does this mean that we're only checking direct dependencies, instead of the dependencies required of our direct dependencies?

Reply:
No, the audit goes through and checks sub-dependencies and gives you a "path" to the dependency with the issue. The root of that path would be a dependency that's in our package.json, so this is restricting it to just ones in `dependencies` vs `devDependencies`.